### PR TITLE
Remove ImageFormat::Invalid

### PR DESCRIPTION
**Pull request by kvark**
Also replaces #2280 

This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2317)

### CODE REVIEW

File: webrender/src/device.rs [hunk 1]
```diff
@@ -1269,7 +1253,7 @@ impl Device {
     pub fn free_texture_storage(&mut self, texture: &mut Texture) {
         debug_assert!(self.inside_frame);
 
-        if texture.format == ImageFormat::Invalid {
+        if texture.width + texture.height == 0 {
```
Comment (glennw) [inline webrender/src/device.rs]: This made me do a double take, but I guess it works :)

Comment (kvark) [reply webrender/src/device.rs]: As far as I see, the only case it tries to handle is when the texture wasn't ever given contents (leaving the size to 0 and the format to `Invalid`).

**Comment by kvark** [comment]
Let me know if you want the version bumped here. Otherwise this should be ready to go.

**Comment by kvark** [comment]
Now that I think of it, this change is effectively a better solution to the flickering problem than the workaround I put it previously. Angle has problems with changing the format, and this PR makes us not even attempt to change the format ever.

**Comment by kvark** [comment]
@bors-servo try
@glennw so I found the problem, and the solution (see last commit) may be concerning. Please take a look.

**Comment by kvark** [comment]
result of IRC discussion: having a hashmap for the free lists of texture cache is not ideal but not really a performance concern, and we can proceed with it.
@bors-servo r=glennw

**Pull request merged**